src/main/java/org/folio/dataexp/service/InputFileProcessor.java

```diff
@@ -72,33 +85,31 @@ private void readCsvFile(FileDefinition fileDefinition, CommonExportFails common
             batch.add(entity);
             duplicatedIds.add(entity.getInstanceId());
           } else {
-            commonExportFails.incrementDuplicatedUUID();
+            commonExportStatistic.incrementDuplicatedUUID();
           }
         } catch (Exception e) {
           log.error("Error converting {} to uuid", id);
-          commonExportFails.addToInvalidUUIDFormat(id);
+          commonExportStatistic.addToInvalidUUIDFormat(id);
         }
         if (batch.size() == BATCH_SIZE_TO_SAVE) {
-          var duplicatedFromDb = findDuplicatedUUIDFromDb(new HashSet<>(batch.stream().map(ExportIdEntity::getInstanceId).toList()), fileDefinition.getJobExecutionId());
-          commonExportFails.incrementDuplicatedUUID(duplicatedFromDb.size());
-          batch.removeIf(e -> duplicatedFromDb.contains(e.getInstanceId()));
-          exportIdEntityRepository.saveAll(batch);
+          insertExportIdService.saveBatch(batch);
+          progress.setReadIds(countOfRead.get());
+          jobExecutionService.save(jobExecution);
           batch.clear();
           duplicatedIds.clear();
         }
       });
     } catch (Exception e) {
-      commonExportFails.setFailedToReadInputFile(true);
+      commonExportStatistic.setFailedToReadInputFile(true);
       log.error("Failed to read for file definition {}", fileDefinition.getId(), e);
     }
-    var duplicatedFromDb = findDuplicatedUUIDFromDb(new HashSet<>(batch.stream().map(ExportIdEntity::getInstanceId).toList()), fileDefinition.getJobExecutionId());
-    commonExportFails.incrementDuplicatedUUID(duplicatedFromDb.size());
-    batch.removeIf(e -> duplicatedFromDb.contains(e.getInstanceId()));
-    exportIdEntityRepository.saveAll(batch);
-  }
+    insertExportIdService.saveBatch(batch);
+    progress.setReadIds(countOfRead.get());
+    jobExecutionService.save(jobExecution);
 
-  private List<UUID> findDuplicatedUUIDFromDb(Set<UUID> ids, UUID jobExecutionId) {
-    return exportIdEntityRepository.findByInstanceIdInAndJobExecutionIdIs(ids, jobExecutionId).stream().map(ExportIdEntity::getInstanceId).toList();
+    int totalExportsIds = (int) exportIdEntityRepository.countByJobExecutionId(jobExecution.getId());
+    int duplicated = countOfRead.get() - totalExportsIds - commonExportStatistic.getDuplicatedUUIDAmount() - commonExportStatistic.getInvalidUUIDFormat().size();
```

Comment:
Can this calculation be extracted to separate method?

Reply:
Hi @khandramai getDuplicatedNumber() was added.